.github/workflows/main-branch.yml

```diff
@@ -6,14 +6,46 @@ on:
 jobs:
   build-test:
     runs-on: ubuntu-latest
+    environment: development
     steps:
       - uses: actions/checkout@v4
       - name: Use Node.js
         uses: actions/setup-node@v4
         with:
           node-version: "22.x"
       - run: npm ci
-      - run: npm run build --if-present
+      - name: Generate random string for Cookie Secret
+        id: generate_secret
+        run: |
+          COOKIE_SECRET=$(openssl rand -base64 32)
+          echo "COOKIE_SECRET=$COOKIE_SECRET" >> $GITHUB_ENV
+
+      - name: Create .env file
```

Comment:
The env file should not be copied into the docker container.
All environment variables should be specified in .github/deployments/templates/deployment.yml